### PR TITLE
update regex in hydration and revsions in GWDM

### PR DESCRIPTION
**Pull request by BSnelling**

### CODE REVIEW

File: hdr_schemata/models/GWDM/v2_0/Revision.py [hunk 1]
```diff
@@ -13,7 +14,7 @@ class Revision(BaseModel):
         **an.version.__dict__,
     )
 
-    url: Url = Field(
+    url: Optional[Url] = Field(
```
Comment (BSnelling) [inline hdr_schemata/models/GWDM/v2_0/Revision.py:17]: This brings the GWDM into alignment with the HDR schema where the Url is optional.

**Review by JamieByrneHDRUK — APPROVED** [review]

File: hdr_schemata/utils/create_markdown.py [hunk 2]
```diff
@@ -219,6 +219,7 @@ def remove_types(data):
 
     form = {}
     form["schema_fields"] = []
+    form["url_regex"] = "(https?:\/\/)?([a-zA-Z0-9-]+\.)+[a-zA-Z]{2,}(:\d+)?(\/[^\s]*)?$"
```
Comment (calmacx) [inline hdr_schemata/utils/create_markdown.py]: regex is forcing `https`, we'd want to allow `http` and maybe `www` ? Think the json-schema is a `uri` format that is accepted

**Review by calmacx — CHANGES_REQUESTED** [review]

File: hdr_schemata/utils/create_markdown.py (hunk 2 above)
Comment (BSnelling) [reply hdr_schemata/utils/create_markdown.py]: The s is optional so `http` or `https` both match. As does a url without the protocol e.g. `www.github.com`.

Comment (BSnelling) [reply hdr_schemata/utils/create_markdown.py]: You make a good point about the `uri` definition in the json-schema.  It allows port numbers among other things, so I've updated the regex to be more flexible. I also checked against url data we have in mk1 to make sure it validates against it too.

File: docs/HDRUK/3.0.0.form.json [hunk 3]
```diff
@@ -1841,5 +1841,6 @@
                   },
                   "location": "omics.platform"
             }
-      ]
+      ],
+      "url_regex": "^\\s*((https?:\\/\\/)*([a-zA-Z0-9-]+\\.?)+[a-zA-Z]{2,}(:\\d+)?(\\/[^\\s]*)?(\n)?)+$"
```
Comment (loki-sinclair-hdruk) [inline docs/HDRUK/3.0.0.form.json:1845]: Tried some rather convoluted urls, and this regex appears to be resilient: https://regex101.com/r/a31iK3/2

**Review by loki-sinclair-hdruk — APPROVED** [review]

**Pull request merged**